### PR TITLE
feat: re-work rln-js

### PR DESCRIPTION
**Pull request by weboko**
In this PR I re-created rln-js with Next.js.
Resolves - https://github.com/waku-org/js-waku-examples/issues/279

Choice for Next.js was because many hackers were using it and it is one of the platforms that fails during development on some of waku errors so I had to investigate it for https://github.com/waku-org/js-waku/issues/1665

This APP supports latest `@waku/rln` API and addresses some fixes. 
Structure of this example can be re-used later for templates. 

UI now looks like 
image

### CODE REVIEW

File: examples/rln-js/README.md [hunk 1]
```diff
@@ -1,30 +1,21 @@
...
-npm run start
-# open  http://127.0.0.1:8080 In your browser
+```bash
+npm run dev
+# or
+yarn dev
+# or
+pnpm dev
+# or
+bun dev
```
[the remark this answers is not in the document]
Comment (weboko) [reply examples/rln-js/README.md:14]: good spot, better to return previous README

File: examples/rln-js/src/services/waku.ts [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+          time: new Date(timestamp).toDateString(),
+        });
+      } catch (error) {
+        console.error("Failed in subscription listener: ", error);
+      }
+    });
+  }
+
+  public addEventListener(name: WakuEventsNames, fn: EventListener) {
+    return this.emitter.addEventListener(name, fn as any);
```
[the remark this answers is not in the document]
Comment (fryorcraken) [reply examples/rln-js/src/services/waku.ts:182]: IMO it's fine in an example

[the remark this answers is not in the document]
Comment (weboko) [reply examples/rln-js/src/services/waku.ts:182]: This `any` should stay as it is consequence of type discrepancy of emitter, event listener and dispatcher

File: examples/rln-js/src/services/rln.ts [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+  CREDENTIALS_FAILURE = "Failed to register credentials, check console",
+}
+
+type EventListener = (event: CustomEvent) => void;
+
+type IRLN = {
+  saveKeystore: () => void;
+  addEventListener: (name: RLNEventsNames, fn: EventListener) => void;
+  removeEventListener: (name: RLNEventsNames, fn: EventListener) => void;
+};
```
[the remark this answers is not in the document]
Comment (weboko) [reply examples/rln-js/src/services/rln.ts:37]: Moved only StatusEventPayload, others makes no sense as they are tightly coupled with rln.ts

**Pull request merged**